client/src/components/organisms/RequestGroupForm.tsx

```diff
@@ -0,0 +1,351 @@
+
+    return "";
+  }
+
+  const getInputRequestTypeNameError = (inputRequestTypeName: string): string => {
+    if (requestTypeNames.find(requestTypeName => requestTypeName === inputRequestTypeName)) {
+      return "There is already a type with this name";
+    }
+    else if (inputRequestTypeName.length > 40) {
+      return "Type name cannot exceed 40 characters";
+    }
+    return "";
+  }
+
+  const getRequestTypeNamesError = (requestTypes: Array<string>): string => {
+    if (requestTypes.length === 0) {
+
+      return "Please create at least 1 type";
+    }
+    return "";
+  }
+
+  const getDescriptionError = (description: string) => {
+    if (!description) {
+      return "Please enter a description";
+    }
+
+    const editorState = EditorState.createWithContent(convertFromRaw(JSON.parse(description)))
+    if (!editorState.getCurrentContent().hasText()) {
+      return "Please enter a description";
+    }
+
+    return "";
+  }
+
+  const getImageError = (image: string) => {
+    if (image === "") {
+      return "Please select an image";
+    }
+    return "";
+  }
+
+  const onNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const newName: string = event.target.value;
+
+    setChangeMade(true);
+    setName(newName);
+    setNameError(getNameError(newName))
+  }
+
+  const onAddRequestType = (newRequestTypeName: string) => {
+    const error = getInputRequestTypeNameError(newRequestTypeName);
+
+    if (error === "") {
+      setRequestTypeNames(requestTypeNames.concat([newRequestTypeName]));
+    }
+    setChangeMade(true);
+    setRequestTypesError(error);
+  }
+
+  const onDeleteRequestType = (targetRequestTypeName: string) => {
+    const newRequestTypeNames = requestTypeNames.filter((requestTypeName) => requestTypeName !== targetRequestTypeName);
+
+    setChangeMade(true);
+    setRequestTypeNames(newRequestTypeNames);
+    setRequestTypesError(getRequestTypeNamesError(newRequestTypeNames))
+  }
+
+  const onInputRequestTypeNameChange = (requestTypeName: string) => {
+    setChangeMade(true);
+    setRequestTypesError(getInputRequestTypeNameError(requestTypeName))
+    return true;
+  }
+
+  const onDescriptionChange = (newDescription: string) => {
+    setDescription(newDescription)
+    setDescriptionError(getDescriptionError(description))
+  }
+
+  const onImageChange = (newImage: string) => {
+    setImageError(getImageError(newImage))
+    if (images.find((imageUrl) => imageUrl === newImage)) {
+      setImage(newImage)
+    }
+  }
+
+  const onSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    const tempNameError = getNameError(name);
+    const tempDescriptionError = getDescriptionError(description);
+    const tempImageError = getImageError(image);
+    const tempRequestTypesError = getRequestTypeNamesError(requestTypeNames);
+
+    if (tempNameError === "" && tempDescriptionError === "" && tempImageError === "" && tempRequestTypesError === "") {
```

Comment:
Just for clarity, maybe it would make it more clear if instead of error === "" everywhere, there was !error

Reply:
will do!